src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -230,13 +248,29 @@ class ListBuffer[A]
   }
 
   def insertAll(idx: Int, elems: IterableOnce[A]): Unit = {
-    ensureUnaliased()
-    val it = elems.iterator
-    if (it.hasNext) {
-      ensureUnaliased()
-      if (idx < 0 || idx > len) throw new IndexOutOfBoundsException(s"$idx is out of bounds (min 0, max ${len-1})")
-      if (idx == len) ++=(elems)
-      else insertAfter(locate(idx), it)
+    if (idx < 0 || idx > len) throw new IndexOutOfBoundsException(s"$idx is out of bounds (min 0, max ${len-1})")
```

Comment:
There was another PR about "out of bounds" which I haven't studied yet, but which lightbend course should I take?
```
scala> ListBuffer(42).insertAll(10, List(27))
java.lang.IndexOutOfBoundsException: 10 is out of bounds (min 0, max 0)
  at scala.collection.mutable.ListBuffer.insertAll(ListBuffer.scala:236)
  ... 32 elided

scala> ListBuffer(42).insertAll(1, List(27))

scala> ListBuffer.empty[Int].tap(_.insertAll(-1, List(42, 27)))
java.lang.IndexOutOfBoundsException: -1 is out of bounds (min 0, max -1)

scala>
```

Reply:
should that be `>= len`? or does passing `len` insert it at the end?

Reply:
```scala
scala> def b[U](op: ListBuffer[Int] => U): ListBuffer[Int] = {
     |   val b = ListBuffer(42)
     |   op(b)
     |   b
     | }
def b[U](op: scala.collection.mutable.ListBuffer[Int] => U): scala.collection.mutable.ListBuffer[Int]

scala> b(_.insertAll(1, List(27)))
val res2: scala.collection.mutable.ListBuffer[Int] = ListBuffer(42, 27)

scala> b(_.insertAll(0, List(27)))
val res3: scala.collection.mutable.ListBuffer[Int] = ListBuffer(27, 42)
```